src/engine/strat_engine/metadata.rs

```diff
@@ -4,262 +4,416 @@
+        try!(f.seek(SeekFrom::Start(0)));
+        try!(f.write_all(&zeroed[..SECTOR_SIZE as usize]));
+        try!(f.write_all(&hdr_buf));
+        try!(f.write_all(&zeroed[(SECTOR_SIZE * 7) as usize..]));
+        try!(f.flush());
+        try!(f.write_all(&hdr_buf));
+        try!(f.write_all(&zeroed[(SECTOR_SIZE * 6) as usize..]));
+        try!(f.flush());
+
+        let regions = try!(MDARegions::initialize(&header, &mut f));
+
+        Ok(BDA {
+            header: header,
+            regions: regions,
+        })
     }
 
-    fn offsets(offset: usize) -> (usize, usize, usize, usize) {
-        (offset + MDA_OFFSETS[0],
-         offset + MDA_OFFSETS[1],
-         offset + MDA_OFFSETS[2],
-         offset + MDA_OFFSETS[3])
+    pub fn load(f: &mut File, header: StaticHeader) -> EngineResult<BDA> {
+        let regions = try!(MDARegions::load(&header, f));
+
+        Ok(BDA {
+            header: header,
+            regions: regions,
+        })
     }
 
-    /// Write out recorded values to buf at offset.
-    pub fn write(&self, buf: &mut [u8], offset: usize) {
-        let (offset1, offset2, offset3, offset4) = MDA::offsets(offset);
+    // Wipe metadata
+    pub fn wipe(f: &mut File) -> EngineResult<()> {
+        let zeroed = [0u8; BDA_STATIC_HDR_SIZE as usize];
 
-        let last_updated = self.last_updated;
-        LittleEndian::write_u64(&mut buf[offset..offset1], last_updated.sec as u64);
-        LittleEndian::write_u32(&mut buf[offset1..offset2], last_updated.nsec as u32);
+        // Wiping Static Header should do it
+        try!(f.seek(SeekFrom::Start(0)));
+        try!(f.write_all(&zeroed));
+        try!(f.flush());
+        Ok(())
+    }
 
-        LittleEndian::write_u32(&mut buf[offset2..offset3], self.used);
-        LittleEndian::write_u32(&mut buf[offset3..offset4], self.crc);
+    /// Save metadata to the disk
+    pub fn save_state(&mut self,
+                      time: &Timespec,
+                      metadata: &[u8],
+                      mut f: &mut File)
+                      -> EngineResult<()> {
+        self.regions.save_state(time, metadata, &mut f)
     }
 
-    /// Read recorded values into buffer at buf_offset.
-    pub fn read(buf: &[u8], buf_offset: usize, offset: SectorOffset) -> MDA {
-        let (offset1, offset2, offset3, offset4) = MDA::offsets(buf_offset);
+    /// Read latest metadata from the disk
+    pub fn load_state(&self, mut f: &mut File) -> EngineResult<Option<Vec<u8>>> {
+        self.regions.load_state(&mut f)
+    }
 
-        MDA {
-            last_updated: Timespec::new(LittleEndian::read_u64(&buf[buf_offset..offset1]) as i64,
-                                        LittleEndian::read_u32(&buf[offset1..offset2]) as i32),
-            used: LittleEndian::read_u32(&buf[offset2..offset3]),
-            crc: LittleEndian::read_u32(&buf[offset3..offset4]),
-            offset: offset,
-        }
+    pub fn last_updated(&self) -> &Option<Timespec> {
```

Comment:
I think the name ```last_updated``` is a bit misleading, both for the name of the method and of the field in the struct. ```last_update_time``` or ```most_recent_update_time``` maybe?